src/de/packet_reader.rs

```diff
@@ -55,6 +55,12 @@ impl<'a> PacketReader<'a> {
             }
         }
 
+        if let Some(packet_length) = self.packet_length {
```

Comment:
I'm not following what went wrong and why this fixes it. Can you add a unit test case that shows what you're observing and also demonstrates that this is now fixed?

Reply:
Thanks for getting back to me. Pushed this out at the end of day, which was not a good idea.
Updated the pull request to correctly fix the issue, and also with a small testcase.